src/Build/BackEnd/Client/MSBuildClient.cs

```diff
@@ -411,7 +416,7 @@ private void HandleServerNodeConsoleWrite(ServerNodeConsoleWrite consoleWrite)
         private void HandleServerNodeBuildResult(ServerNodeBuildResult response)
         {
             CommunicationsUtilities.Trace("Build response received: exit code {0}, exit type '{1}'", response.ExitCode, response.ExitType);
-            _exitResult.MSBuildClientExitType = MSBuildClientExitType.Success;
+            _exitResult.MSBuildClientExitType = _buildCanceled ? MSBuildClientExitType.Cancelled : MSBuildClientExitType.Success;
```

Comment:
To resolve partially situation 2 from my other comment about race condition, should we here take into account the `response.ExitType` instead of `_buildCanceled`? There should be cancellation in the enum, I believe.

Reply:
I checked the `ExitType` enum and unfortunately, it doesn't contain value for canceled/aborted build. We could add this value, but I guess it would require to find a place or places where we should identify this condition and emit correct value.

Reply:
Hm, if there is no Cancellation status, then obviously there is no distinction between cancelled and failed builds. One of our options is also do not have this distinction. See, actually we do not need to know that the build was cancelled, as there is no change in behavior in this case.

Reply:
Yes, So I suggest to drop this value from this enum, since we do not have yet any logic around it.